File: src/main/java/com/everyonewaiter/global/annotation/ApiErrorResponse.java
package com.everyonewaiter.global.annotation;

import com.everyonewaiter.domain.shared.ErrorCode;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;
import org.springframework.http.MediaType;

@Target(ElementType.METHOD)
@Retention(RetentionPolicy.RUNTIME)
public @interface ApiErrorResponse {

  String summary() default "";

  ErrorCode code();

  String exampleName() default "";

  String mediaType() default MediaType.APPLICATION_JSON_VALUE;

}

File: src/main/kotlin/com/everyonewaiter/global/annotation/ApiErrorResponse.kt
package com.everyonewaiter.global.annotation

import com.everyonewaiter.global.exception.ErrorCode
import org.springframework.http.MediaType

@Target(AnnotationTarget.FUNCTION)
@Retention(AnnotationRetention.RUNTIME)
annotation class ApiErrorResponse(
    val code: ErrorCode,
    val summary: String = "",
    val exampleName: String = "",
    val exampleMessage: String = "",
    val mediaType: String = MediaType.APPLICATION_JSON_VALUE,
)
